### DISCUSSION TITLE
Improved TRIPLET settings for install command?

### DISCUSSION THREAD

**Discussion by YuHuanTin** [microsoft/vcpkg#35068] [Q&A] [answered]
Intuitively I would think that both `VCPKG_DEFAULT_TRIPLET` and `VCPKG_DEFAULT_HOST_TRIPLET` should have been specified when `install <xxx>:x64-mingw-dynamic` was used rather than needing to be specified in the environment variable

Or is there a better way to specify `mingw-dynamic` without using an environment variable to set it?

**Comment by dg0yt** [comment, marked as answer]
What's wrong with environment variables for overriding defaults?
You can pass `--triplet=... --host-triplet=...` to override defaults and environment per invocation.
`ìnstall xxx:<triplet>` overrides per package.

Note that there is  an order here: defaults - environment - invocation - package

And there is probably no desire to have anything special for mingw. (And it is special, because mingw is normally used when MSVC isn't available at all.)

**Reply by YuHuanTin** [reply]
`vcpkg install <xxx> --triplet=x64-mingw-dynamic --host-triplet=x64-mingw-dynamic`
It seems like a good idea to be able to set each package without changing the environment variable 👍
